docs/_newsfragments/rich.feature.rst

```diff
@@ -0,0 +1 @@
+Added ``__rich__`` method to the ``Result`` class for better integration with the Rich library. 
```

Comment:
We should link to the class here, it is otherwise unclear which class we mean here.